pandas/tests/indexes/datetimes/test_tools.py

```diff
@@ -101,6 +101,32 @@ def test_to_datetime_format_YYYYMMDD(self, cache):
         expected = Series(["20121231", "20141231", "NaT"], dtype="M8[ns]")
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "input_s",
+        [
+            # None with Strings
+            Series(["19801222", None, "20010112", np.nan, "NaT", pd.NaT]),
+            # None with Integers
+            Series([19801222, None, 20010112, np.nan, "NaT", pd.NaT]),
+        ],
+    )
+    def test_to_datetime_format_YYYYMMDD_with_none(self, input_s):
+        # GH 30011
+        # format='%Y%m%d'
+        # with None
+        expected = Series(
+            [
+                Timestamp("19801222"),
```

Comment:
no, your expected is just pd.NaT for *any* null values

Reply:
Thanks @jreback for your constant review and help. Made the changes accordingly.